lib/pf/api/unifiedapiclient.pm

```diff
@@ -183,7 +191,11 @@ sub call {
     $curl->setopt(CURLOPT_CUSTOMREQUEST, $method);
 
     if($self->token) {
-        $curl->setopt(CURLOPT_HTTPHEADER, ["Authorization: Bearer ".$self->token]);
+        $curl->pushopt(CURLOPT_HTTPHEADER, ["Authorization: Bearer ".$self->token]);
+    }
+
+    if(defined($self->tenant_id)) {
```

Comment:
We should use the global tenant_id if this is undefined to avoid having to change all places where we use pf::api::unifiedapiclient.

Reply:
If you don't specify the header, it makes it default to the tenant ID associated to the token.

This piece of code handles the case where its not specified:
https://github.com/inverse-inc/packetfence/blob/devel/go/api-frontend/aaa/authorization.go#L241-L255

Its better to only specify the header to the api-frontend when its needed than to include it at all time when performing backend work

Reply:
This would mean every place we use the ```pf::api::unifiedapiclient``` we must explicitly define which tenant_id we are using when 99.9% of the case we would be using the current globally defined one.

It is pretty reasonable to assume if we are making an API call then we should the current globally defined one.

By not using the globally defined one it just makes it easier to make to forget to use.

Reply:
It uses the tenant associated to the credentials if its not specified.

Since the credentials that are set in the client by default are the ones of the system user (associated to the global tenant), so by using it as we always do, it already does what you describe (i.e. using the global tenant) without specifying the tenant header

Reply:
I understand the miscommunication the 'global tenant' is not the one in the database.
I meant the globally defined variable ```$pf::config::tenant::CURRENT_TENANT```

Reply:
I see, we were indeed talking of 2 completely different things

Your idea is good to prevent duplication of efforts